tools/lkl/lib/hijack/init.c

```diff
@@ -181,7 +181,7 @@ int lkl_debug, lkl_running;
 static int nd_id = -1;
 static struct lkl_netdev *nd;
 
-void __attribute__((constructor(102)))
+void __attribute__((constructor))
```

Comment:
DPDK initialization should be completed before `lkl_netdev_dpdk_create()` is called.  So `hijack_init` should be called after the initialization.

and the most of initialization (of DPDK) are declared as `__attribute__((constructor))`, which the user of DPDK (i.e., LKL) can only control this order by the order of linker argument (AFAIK).

So the HOOK defines should be fine as-is since the defines are safe even if those are executed before the DPDK initialization.